spark/src/main/scala/org/apache/spark/sql/comet/CometRowToColumnarExec.scala

```diff
@@ -60,8 +62,17 @@ case class CometRowToColumnarExec(child: SparkPlan)
     val timeZoneId = conf.sessionLocalTimeZone
     val schema = child.schema
 
-    child
-      .execute()
+    val rdd: RDD[InternalRow] = if (child.supportsColumnar) {
+      child
+        .executeColumnar()
+        .mapPartitionsInternal { iter =>
+          iter.flatMap(_.rowIterator().asScala)
+        }
+    } else {
+      child.execute()
+    }
```

Comment:
My understanding of this is that it allows us to read from parquet files containing structs in tests by falling back to Spark, and that this new code wouldn't be executed outside of tests? Could you add some comments in here to that effect if this is the case?

Reply:
I mean I would say this is no more-or-less just for testing than the existing use cases for CometRowToColumnarExec. I was using this for real queries while testing this out on real data. It could be a legitimate use depending on when complex type support is added to the parquet readers.